test/integration/testExecute.js

```diff
@@ -431,5 +431,88 @@ describe('Execute test - variant', function () {
 
     it(testCase.name, createItCallback(testCase, rowAsserts));
   });
-});
 
+  describe.only( 'connection.execute() Resubmitting requests using requestId and different connections', function () {
+    const createTable = 'create or replace table test_request_id(colA string)';
+    let  firstConnection;
+    let  secondConnection;
+    before(async () => {
+      firstConnection = testUtil.createConnection();
+      secondConnection = testUtil.createConnection();
+      await testUtil.connectAsync(firstConnection);
+      await testUtil.connectAsync(secondConnection);
+      await testUtil.executeCmdAsync(firstConnection, 'truncate table if exists test_request_id');
+      await testUtil.executeCmdAsync(firstConnection, createTable);
+    });
+
+    beforeEach(async () => {
+      await testUtil.executeCmdAsync(firstConnection, 'truncate table if exists test_request_id');
+    });
+
+    after(async () => {
+      await testUtil.executeCmdAsync(firstConnection, 'drop table if exists test_request_id');
+    });
+
+    it('Do not INSERT twice when the same request id and connection', async () => {
+      let requestId;
+      await testUtil.executeCmdAsyncWithAdditionalParameters(firstConnection, 'INSERT INTO test_request_id VALUES (\'testValue\');')
+        .then((result) => {
```

Comment:
why do we mix await with promises? can we have these tests 100% async-await?

Reply:
Sure. Fixed